### PR TITLE
added a test_dashboard.py file in the tests folder

### LINKED ISSUE #79: Add test cases for `dashboard` endpoints:
1. Read guideline [here](https://gist.github.com/djeck1432/f722f5fe518c80b248ff609627da25e5)
2. Set up the project, read [README.md](https://github.com/djeck1432/spotnet/blob/main/README.md#development-environment-setup) how to do that.
3. Create test cases (positive and negative scenarios) for dashboard endpoints which you can find [here](https://github.com/djeck1432/spotnet/blob/main/web_app/api/dashboard.py#L20C11-L20C24)
P.s. Database connection has been already mocked, so you don't need to take care of it.

### PR DESCRIPTION
**Pull request by PoulavBhowmick03**
Fixes #79 

All 7 test cases are running

### CODE REVIEW

File: web_app/tests/test_dashboard.py [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+
+        mock_get_contract_address_by_wallet_id.return_value = None
+        mock_get_positions_by_wallet_id.return_value = []
+        mock_get_wallet_balances.return_value = {"ETH": 5.0, "USDC": 1000.0}
+        mock_get_zklend_position.return_value = {"products": []}
+
+        async with AsyncClient(app=app, base_url="http://test") as ac:
+            response = await ac.get(f"/api/dashboard?wallet_id={wallet_id}")
+
+        assert response.status_code == 200
```
Comment (djeck1432) [inline web_app/tests/test_dashboard.py:146]: use response.ok

File: web_app/tests/test_dashboard.py [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+        "web_app.contract_tools.mixins.dashboard.DashboardMixin.get_zklend_position",
+        new_callable=AsyncMock,
+    ) as mock_get_zklend_position:
+
+        mock_get_contract_address_by_wallet_id.return_value = None
+        mock_get_positions_by_wallet_id.return_value = []
+        mock_get_wallet_balances.return_value = {"ETH": 5.0, "USDC": 1000.0}
+        mock_get_zklend_position.return_value = {"products": []}
+
+        async with AsyncClient(app=app, base_url="http://test") as ac:
```
Comment (djeck1432) [inline web_app/tests/test_dashboard.py:143]: Use constant for `base_url` to not repeat yourself

**Review by djeck1432 — CHANGES_REQUESTED** [review]

**Comment by djeck1432** [comment]
@PoulavBhowmick03 Please, make sure that you don't have any issues with pylint and test running workflow

**Comment by djeck1432** [comment]
@PoulavBhowmick03 https://github.com/djeck1432/spotnet/actions/runs/11517730577/job/32063064862?pr=101
fix please pylint issue

**Comment by PoulavBhowmick03** [comment]
@djeck1432 PTAL. All tests are running and fixed the issues
PTAL

**Review by djeck1432 — APPROVED** [review]
Looks good to me

**Comment by PoulavBhowmick03** [comment]
Thanks! Looking forward to contributing to more issues

**Pull request merged**